src/sdam/server_description.ts

```diff
@@ -179,15 +180,15 @@ export class ServerDescription {
    * Determines if another `ServerDescription` is equal to this one per the rules defined
    * in the {@link https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#serverdescription|SDAM spec}
    */
-  equals(other: ServerDescription): boolean {
+  equals(other?: ServerDescription | null): boolean {
     const topologyVersionsEqual =
-      this.topologyVersion === other.topologyVersion ||
-      compareTopologyVersion(this.topologyVersion, other.topologyVersion) === 0;
+      this.topologyVersion === other?.topologyVersion ||
```

Comment:
this comparison feels like it violates this portion in the `compareTopologyVersion` logic:

> If the response topologyVersion is unset or the ServerDescription's topologyVersion is null, the client MUST assume the response is more recent.

why do we check strict equality here instead of just using the outcome of `compareTopologyVersion` from the outset? are we explicitly trying to catch `null === null`?

Reply:
Thanks for calling this out, when I made the change to respect that part of the spec we get test failures so something is amiss here. I think that the equality case and the comparison test have different rules about nullishness. I'll file a ticket.

NODE-4510